### PR TITLE
Change the wording in `std::fmt::Write::write_str`

### PR DESCRIPTION
**Pull request by tbu-**
Refer to the error instead of expanding its name.

### CODE REVIEW

**Comment by m-ou-se** [comment]
The full path `std::fmt::Error` seems useful to show, to distinguish it from `std::io::Error` (which is used by the very similar `std::io::Write` trait).

**Comment by tbu-** [comment]
> The full path `std::fmt::Error` seems useful to show, to distinguish it from `std::io::Error` (which is used by the very similar `std::io::Write` trait).

Added the path to the line above.

File: library/core/src/fmt/mod.rs [hunk 1]
```diff
@@ -112,9 +112,9 @@ pub trait Write {
     ///
     /// # Errors
     ///
-    /// This function will return an instance of [`Error`] on error.
+    /// This function will return an instance of [`std::fmt::Error`] on error.
```
Comment (dtolnay) [inline library/core/src/fmt/mod.rs:115]: This should unbreak the doc build.

```suggestion
    /// This function will return an instance of [`std::fmt::Error`][Error] on error.
```

**Review by dtolnay — CHANGES_REQUESTED** [review]

**Review by dtolnay — APPROVED** [review]
Thanks!

**Pull request merged**